api/test/trace/trace_state_test.cc

```diff
@@ -60,81 +61,129 @@ TEST(EntryTest, SetValue)
 
 // -------------------------- TraceState class tests ---------------------------
 
-TEST(TraceStateTest, DefaultConstruction)
+std::string create_ts_return_header(std::string header)
 {
-  TraceState s;
-  opentelemetry::nostd::string_view return_val = "";
-  EXPECT_FALSE(s.Get("missing_key", return_val));
-  EXPECT_EQ(return_val, "");
-  EXPECT_TRUE(s.Empty());
-  EXPECT_EQ(s.Entries().size(), 0);
+  TraceState ts = TraceState::FromHeader(header);
+  return ts.ToHeader();
 }
 
-TEST(TraceStateTest, Set)
+std::string header_with_max_members()
 {
-  TraceState s;
-  opentelemetry::nostd::string_view key = "test_key";
-  opentelemetry::nostd::string_view val = "test_value";
-  s.Set(key, val);
+  std::string header = "";
+  auto max_members   = opentelemetry::trace::TraceState::kMaxKeyValuePairs;
+  for (int i = 0; i < max_members; i++)
+  {
+    std::string key   = "key" + std::to_string(i);
+    std::string value = "value" + std::to_string(i);
+    header += key + "=" + value;
+    if (i != max_members - 1)
+    {
+      header += ",";
+    }
+  }
+  return header;
+}
+
+TEST(TraceStateTest, ValidateHeaderParsing)
+{
+  std::string trace_state_header = "k1=v1";  // valid header
+  EXPECT_EQ(create_ts_return_header(trace_state_header), trace_state_header);
+
+  trace_state_header = "K1=V1";  // invalid header, key can't start with uppercase.
+  EXPECT_EQ(create_ts_return_header(trace_state_header), "");
 
```

Comment:
```suggestion
  struct {
    const char* input;
    const char* expected;
  } testcases[] = {
    { "k1=v1", "k1=v1" },
    { "K1=V1", "" },
    { "k1=v1,k2=v2,k3=v3", "k1=v1,k2=v2,k3=v3" },
    { "k1=v1,k2=v2,,", "k1=v1,k2=v24" }
    // not all cases covered
  };

  for (auto& testcase : testcases)
  {
    EXPECT_EQ(TraceState::FromHeader(testcase.input).ToHeader(), testcase.expected);
  }
```
Also just a nit. I think this makes it easier to read and to add tests.

Reply:
Thanks.